zcash_primitives/src/transaction/builder.rs

```diff
@@ -229,6 +275,48 @@ impl<'a, P: consensus::Parameters, R: RngCore> Builder<'a, P, R> {
         }
     }
 
+    /// Adds an Orchard note to be spent in this bundle.
+    ///
+    /// Returns an error if the given Merkle path does not have the required anchor for
+    /// the given note.
+    pub fn add_orchard_spend<FR: FeeRule>(
+        &mut self,
+        sk: orchard::keys::SpendingKey,
+        note: orchard::Note,
+        merkle_path: orchard::tree::MerklePath,
+    ) -> Result<(), Error<FR::Error>> {
```

Comment:
Hmm, the other adders (`add_sapling_spend` etc) return the underlying protocol-specific errors, instead of mapping to `builder::Error`. We should probably do the same here for consistency, which would enable removing the `FR` generic parameter.

Reply:
We can't, because we're returning `Error::OrchardAnchorNotAvailable`.

Reply:
However, this should be universally quantified without a bound.